dts/arm/st/f4/stm32f401.dtsi

```diff
@@ -37,6 +37,16 @@
 			reg = <0x40003800 0x400>;
 			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x00004000>;
 			interrupts = <36 5>;
+			dma = <&dma1 4 0 STM32_DMA_MEMORY_TO_PERIPH
```

Comment:
Is it applicable to all boards that will use this SoC? Or is it board/application dependent?
In the latter case this should be done either at board level, either as an overlay in matching sample.

Reply:
Which DMA instance and which channel to use is applicable to all boards using this SoC. And other configurations are depends on the I2S driver. I retrieved them from them old I2S driver when all of them are hard coded in the I2S driver. So at least every board using our stm32 i2s driver in zephyr should use this configuration.

Reply:
Which DMA instance and which channel to use is applicable to all boards using this SoC. And other configurations are depends on the I2S driver. I retrieved them from them old I2S driver when all of them are hard coded in the I2S driver. So at least every board using our stm32 i2s driver in zephyr should use this configuration.

Reply:
@avisconti , can you comment here?
Is this configuration really SoC specific: applies for all socs (and socs derived from this one), on all boards, for all use cases.
For instance, if I don't use I2S driver, can't I reuse this channel for other purpose?

Reply:
@erwango @cybertale 
Well, some time has passed by since I have explored the DMA so I need to reconstruct all the knowledge I had.
Anyway, every SoC defines its own table of dma-peripheral connections which is fixed of course, because it reflects the way the peripheral is connected to the DMA. So, for example, the DMA1 ch0 strm3 is connected to SPI2_Rx (I2S2_Rx) and DMA1 ch0 strm4 is connected to SPI2_Tx (I2S2_Tx).

So, the combination channel + stream is fixed to a particular peripheral. Here, it is defined for the I2S instance the DMA configuration.

Reply:
I think this work is a huge tentative to move DMA information out from the I2S driver and move it to dts.
This can be potentially an example for other drivers too that wants to make use of DMA.

Reply:
> Is it applicable to all boards that will use this SoC? Or is it board/application dependent?
> In the latter case this should be done either at board level, either as an overlay in matching sample.

The DMA to Peripheral connections are hardwired in the SoC and described in a table in SoC Reference Manual. So, it will be same for all boards using that SoC.

Reply:
ok, thanks for clarification. Closing this point.